File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    length = len(progresses)
    flag = [False] * length
    cnt = 0
    
    while cnt < length:
        for i in range(length):
            if progresses[i] < 100:
                progresses[i] += speeds[i]
                if progresses[i] >= 100:
                    flag[i] = True
        
        for i in range(cnt, length):
            if flag[i] == False:
                if i != cnt:
                    answer.append(i-cnt)
                    cnt = i
                break
        else:
            answer.append(length-cnt)
            cnt = length
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int n = progresses.size();
    int cnt = 0;
    
    while (cnt < n){
        for (int i=cnt; i<n; i++){
            progresses[i] += speeds[i];
        }
        
        if (progresses[cnt] >= 100){
            int cur = cnt + 1;
            while (cur < n && progresses[cur] >= 100){
                cur++;
            }
            answer.push_back(cur-cnt);
            cnt = cur;
        }
    }
    
    return answer;
}